src/buy/components/BuyDropdown.tsx

```diff
@@ -0,0 +1,110 @@
+import { useOnchainKit } from '@/core-react/useOnchainKit';
+import { openPopup } from '@/ui-react/internal/utils/openPopup';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useAccount } from 'wagmi';
+import { getRoundedAmount } from '../../core/utils/getRoundedAmount';
+import { ONRAMP_BUY_URL } from '../../fund/constants';
+import { getFundingPopupSize } from '../../fund/utils/getFundingPopupSize';
+import { background, cn, color, text } from '../../styles/theme';
+import { ONRAMP_PAYMENT_METHODS } from '../constants';
+import { BuyOnrampItem } from './BuyOnrampItem';
+import { useBuyContext } from './BuyProvider';
+import { BuyTokenItem } from './BuyTokenItem';
+
+export function BuyDropdown() {
+  const { projectId } = useOnchainKit();
+  const { to, fromETH, fromUSDC, from, startPopupMonitor, setIsDropdownOpen } =
+    useBuyContext();
+  const { address } = useAccount();
+
+  const handleOnrampClick = useCallback(
+    (paymentMethodId: string) => {
+      return () => {
+        const assetSymbol = to?.token?.symbol;
+        let fundAmount = to?.amount;
+        // funding url requires a leading zero if the amount is less than 1
+        if (fundAmount?.[0] === '.') {
+          fundAmount = `0${fundAmount}`;
+        }
+        const fundingUrl = `${ONRAMP_BUY_URL}/one-click?appId=${projectId}&addresses={"${address}":["base"]}&assets=["${assetSymbol}"]&presetCryptoAmount=${fundAmount}&defaultPaymentMethod=${paymentMethodId}`;
+        const { height, width } = getFundingPopupSize('md', fundingUrl);
+        const popupWindow = openPopup({
+          url: fundingUrl,
+          height,
+          width,
+          target: '_blank',
+        });
+
+        if (popupWindow) {
+          // Detects when the popup is closed
+          // to stop loading state
+          startPopupMonitor(popupWindow);
+        }
+      };
+    },
+    [address, to, projectId, startPopupMonitor],
+  );
+
+  const formattedAmountUSD = useMemo(() => {
+    if (!to?.amountUSD || to?.amountUSD === '0') {
+      return null;
+    }
+    const roundedAmount = Number(getRoundedAmount(to?.amountUSD, 2));
+    return `$${roundedAmount.toFixed(2)}`;
+  }, [to?.amountUSD]);
+
+  const isToETH = to?.token?.symbol === 'ETH';
+  const isToUSDC = to?.token?.symbol === 'USDC';
+  const showFromToken =
+    to?.token?.symbol !== from?.token?.symbol &&
+    from &&
+    from?.token?.symbol !== 'ETH' &&
+    from?.token?.symbol !== 'USDC';
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        setIsDropdownOpen(false);
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
```

Comment:
We also repeat this functionality a lot throughout our components. I'm going to look into creating a reusable hook